posthog/models/property_definition.py

```diff
@@ -30,3 +30,7 @@ class Meta:
 
     def __str__(self) -> str:
         return f"{self.name} / {self.team.name}"
+
+    # This is a dynamically calculated field in api/property_definition.py. Defaults to `True` here to help serializers.
+    def is_event_property(self) -> bool:
+        return True
```

Comment:
So actually how about treating this prop as `boolean | null` and then if you don't pass `eventNames` we send all as `null`, that way there's a distinction between seen, not seen and not applicable.